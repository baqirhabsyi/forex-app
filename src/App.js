import React, { Component } from 'react';
import { CurrencyInput, CurrencyList } from './layout'
import TextInputHandler from './core/handler/TextInputHandler'
import AddCurrencyComponent from './components/AddCurrencyComponent'
import AddButtonHandler from './core/handler/AddButtonHandler'
import CurrencyListHandler from './core/handler/CurrencyListHandler'
import './App.css';

const initialState = {
  fetch: true,
  value: 1.00,
  addCurrencyValue: '',
  rates: {},
  base: '',
  mainCurrency: 'USD',
  currencyList: [],
}

class App extends Component {
  constructor(props) {
    super(props)
    this.state = initialState

    Object.assign(this, TextInputHandler)
    Object.assign(this, AddButtonHandler)
    Object.assign(this, CurrencyListHandler)

    this.onChange = this.onChange.bind(this)
    this.onAddCurrencyChange = this.onAddCurrencyChange.bind(this)
    this.addCurrency = this.addCurrency.bind(this)
    this.removeListItem = this.removeListItem.bind(this)
    this.fetchData = this.fetchData.bind(this)
  }

  componentDidMount() {
    this.fetchData()
  }

  render() {
    return (
      (this.state.fetch ?
        <div>
          Loading...
        </div> :
        <div className="App">
          <CurrencyInput value={this.state.value} onChange={this.onChange} currencyName="USD - United States Dollar" currencyCode={this.state.mainCurrency} />
          <CurrencyList data={this.state.currencyList} from={this.state.mainCurrency} value={this.state.value} rates={this.state.rates} onPressMinus={this.removeListItem} ownState={this} />
          <AddCurrencyComponent value={this.state.addCurrencyValue} text="Add More Currency" onSubmit={(currencyState) => this.addCurrency(this.state.addCurrencyValue, this.state.rates, currencyState, this.state.currencyList)} onChange={this.onAddCurrencyChange} />
        </div>
      )
    );
  }
}

export default App;
