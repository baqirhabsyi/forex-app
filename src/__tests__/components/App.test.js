import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../../App'
import ReactDOM from 'react-dom'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App displays loading text for the first time', () => {
  describe('render()', () => {
    test('renders the loading text', () => {
      const component = shallow(<App />)

      expect(toJson(component)).toMatchSnapshot()
    })
  })
})


