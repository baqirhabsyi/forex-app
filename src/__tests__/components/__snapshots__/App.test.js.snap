// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the loading text 1`] = `
<div>
  Loading...
</div>
`;
